### DISCUSSION TITLE
[11.0.0-preview2] Simple navigation system broken

### DISCUSSION THREAD

**Discussion by Mrcubix** [AvaloniaUI/Avalonia#9134] [Q&A] [answered]
I prefer creating a discussion first in case this is all due to skill issue and some changes i am not aware of.

Here is the basic example repo i'm using for this : https://github.com/Mrcubix/Avalonia-Router-Test

The App is simple, first view contains a simple button to move to a list view.

In 11.0.0-preview1:

The app work properly, you can click the button and the view change.

In 11.0.0-preview2:

The button is greyed out for some reasons.
moretheless, Any attempt at passing an ICommand or ReactiveCommand or a call on the UI Thread will result in an exception which:

- Doesn't get displayed properly in vscode (the exception doesn't show up at all and you have to check the debugger instead)
- Contains "Call from an invalid thread"

## Update

It seems as you cannot access Parent class Properties of a DataContext, which is odd,
I have to implement a method that has no arguments in my view model, or else, it cannot be called from the UI which is very weird

```cs
// work
public void ShowView()
{
    Remote.ShowView("Example-List");
}

// also work
public bool ShowViewAndReturn()
{
    return Remote.ShowView("Example-List");
}

// greyed out
// <Button Command="{Binding ShowViewAndReturn}" CommandParameter="Example-List" Margin="0,10,0,0" HorizontalAlignment="Center" VerticalAlignment="Center">Go to List Page 0</Button>
public bool ShowViewAndReturn(string view)
{
    return Remote.ShowView(view);
}
```

there was also a warning about ReactiveUI running in .NET Standard? asking me to download the .NET Standard package when my project is targetting .NET 6

**Comment by timunie** [comment]
Ah I see. You have two times the same Method with different notation. This will not work. Use one with parameter and set null for example if no parameter is present. 

Or even better use an ICommand where you have much more control over what is going on.

**Reply by Mrcubix** [reply]
This is obviously not the issues, these methods were tested 1 by 1, they were never in my code at the same time

Edit: on top of that, it doesn't seem like you read this in its entirety as i mentionned that you cannot access parent class properties of a DataContext anymore

**Reply by timunie** [reply]
Looking at you code I think this will not work as a command parameter: 

```xml
 CommandParameter="{Binding PageNumber} - 1">
```

If you want to add or subtract things, you need a Converter.

**Reply by Mrcubix** [reply]
It's not the issue either, as a reminder, it work in 11.0.0-preview1 as you may or may not have have tried previously

Moretheless, i did state in the comment of the last Method tested that this does not work, and is greyed out, probably because of the presence of CommandParameter.

i will test this even further in a couple mins once i get my things in my apartement

**Reply by timunie** [reply]
Sorry, I have not had time to test it yet. Nevertheless I wanted to give you some ideas of a possible root cause. If you don't want me to do that, please let me know. I'll not answer anymore in that case unless I really had more time to fix the issue on my own. 

Remember it's my free time me spending to support you. 

Happy coding,

Tim

**Reply by Mrcubix** [reply]
I am aware of that as i've been a volunteer of my own in the past,

It's just frustrating to read such answers when you know the amount of time you spent testing such thing

Edit: i'm also getting deep into another project and i'm slowly digging out more issues that aren't related to my code and have to start using a bunch of workarounds,
Problems which are still not fixed or made worse 400 commits later in preview 2

I would want to say that some of these issues may not be caused by avalonia but unfortunately, Avalonia's nuget feed is down (502 bad gateway) so i can't debug anything

**Comment by maxkatz6** [comment, marked as answer]
1. `CommandParameter="{Binding PageNumber} - 1"` syntax is not supported and never was. 
2. `public bool ShowViewAndReturn(string view)` [after that PR](https://github.com/AvaloniaUI/Avalonia/pull/8905) only supported syntax is either method with no parameters or a method with a single object parameter. I believe this is a problem with your code.

That PR was a breaking change, but needed to avoid other problems with binding to methods syntax. @Takoooooo @grokys

**Reply by maxkatz6** [reply]
> there was also a warning about ReactiveUI running in .NET Standard? asking me to download the .NET Standard package when my project is targetting .NET 6

This one is weird, as ReactiveUI has NET6 target for sure. But it doesn't seem to be problem in Avalonia, but rather in SDK.

**Comment by timunie** [comment]
Okay I had time to take a deeper look. It works for me if I enable compiled bindings. (I do this always as they help to debug bindings). But that should not be needed tbh. So for me it's a bug. See:

https://github.com/Mrcubix/Avalonia-Router-Test/pull/2 

@maxkatz6 @Takoooooo @grokys what is your opinion? 

P.S: If you want to learn more about CompiledBidnings see: https://docs.avaloniaui.net/docs/data-binding/compiledbindings
